ocaml/src/albamgr_client.ml

```diff
@@ -509,10 +512,50 @@ let wrap_around (ara_c:Arakoon_client.client) =
      Llio.input_string ic >>= fun msg ->
      Lwt.fail (Arakoon_exc.Exception (rc, msg))
 
-let wrap_around' ara_c =
-  wrap_around ara_c >>= fun c ->
+let wrap_around' ?(consistency=Arakoon_client.Consistent) ara_c =
+  wrap_around ~consistency ara_c >>= fun c ->
   Lwt.return (new client (c :> basic_client))
 
+type refresh_result =
+  | Retry
+  | Res of Albamgr_protocol.Protocol.Arakoon_config.t
+
+let retrieve_cfg_from_any_node current_config =
+    Lwt_log.debug "retrieve_from_any_node" >>= fun () ->
+    let cluster_id, node_hashtbl = current_config in
+    let node_names = Hashtbl.fold (fun nn _  acc -> nn :: acc) node_hashtbl [] in
+
+    let rec loop = function
+      | [] -> Lwt.return Retry
+      | node_name :: rest ->
+         Lwt.catch
+           (fun  () ->
+            let node_cfg  =
+              Hashtbl.find node_hashtbl node_name
+            in
+            let open Albamgr_protocol.Protocol.Arakoon_config in
+            let cfg = from_node_client_cfg node_cfg in
+            Lwt_log.debug_f "retrieving from %s" node_name >>= fun () ->
+            Client_helper.with_client'
```

Comment:
we were investigating a but and the addition was done under that hypothesis. IIRC.